### PR TITLE
feat: auto-discover skills/commands from Claude Code, Codex, and Gemini configs

### PR DESCRIPTION
**Pull request by kulvirgit**
## Summary
- Adds auto-discovery of skills and commands from external AI tool configs (Claude Code `.claude/commands/*.md`, Codex `.codex/skills/**/SKILL.md`, Gemini `.gemini/skills/**/SKILL.md` and `.gemini/commands/*.toml`)
- Discovered skills are additive — existing skills always take precedence
- Controlled by `experimental.auto_skill_discovery` config toggle (default: `true`)
- Fixes skill-backed commands to expose `$ARGUMENTS` hints via `hints(skill.content)`

## Test plan
- [x] `bun turbo typecheck` passes
- [x] 14 new tests in `test/skill/discover-external.test.ts` pass
- [x] Existing `test/skill/skill.test.ts` (13 tests) — no regressions
- [x] Existing `test/tool/skill.test.ts` (5 tests) — no regressions
- [ ] Manual: create `~/.claude/commands/test-cmd.md` → verify `/test-cmd` appears
- [ ] Manual: create `~/.gemini/commands/deploy.toml` → verify `/deploy` discovered
- [ ] Manual: set `experimental.auto_skill_discovery: false` → verify no discovery

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Optional startup auto-discovery of external skills/commands (experimental, off by default).
  * Interactive discovery flow and tool to list and add discovered skills, plus a built-in discover command/template and registry integration.

* **Bug Fixes**
  * Skill-derived commands now surface prompt-derived hints correctly.

* **Tests**
  * Added end-to-end tests and a sanity check covering discovery, formats, deduplication, edge cases, and loading.

### CODE REVIEW

File: packages/opencode/src/skill/skill.ts [hunk 1]
```diff
@@ -230,6 +230,26 @@ export namespace Skill {
...
+            dirs.add(path.dirname(skill.location))
+            added.push(skill.name)
+          }
+        }
+        setSkillDiscoveryResult(added, sources)
+      } catch (error) {
+        log.error("external skill discovery failed", { error })
+      }
+    }
+    // altimate_change end
```
[the remark this answers is not in the document]
Comment (kulvirgit) [reply packages/opencode/src/skill/skill.ts]: Fixed: removed the set/consume pattern. Discovery results are now surfaced via the `skill_discover` tool (`/discover-skills` command) which lists and adds skills on demand.
